packages/browser-agent-core/common/util/obfuscate.js

```diff
@@ -0,0 +1,80 @@
+import { getConfigurationValue } from '../config/config'
+import { SharedContext } from '../context/shared-context'
+import { protocol } from '../url/protocol'
+
+var fileProtocolRule = {
+  regex: /^file:\/\/(.*)/,
+  replacement: 'file://OBFUSCATED'
+}
+
+let recordedSupportability = false
+
+export class Obfuscator extends SharedContext {
+  constructor(parent) {
+    super(parent) // gets any allowed properties from the parent and stores them in `sharedContext`
+
+    /* Note: we may need these lines from legacy code -- (when passing in an optional metrics Instrument instance as argument)
+    if (shouldObfuscate()) metrics.recordSupportability('Generic/Obfuscate/Detected')
```

Comment:
These should be re-implemented before release

Reply:
This is implemented in the metrics feature